### DISCUSSION TITLE
Configuration of devices

### DISCUSSION THREAD

**Discussion by jusschwitalla** [QUARK-framework/QUARK#23] [Ideas]
Hi there,

I would like to implement devices which have some parameters. For example a device could be some parametrized QPU simulator. Therefore I suggest that devices should get a  "get_parameter_options" method like the other QUARK components. The benchmark manager could ask the user to provide the corresponding values (the same way as it does it for the solver for instance) and than could pass the selected parameter values as a device_config object to the solvers run method.

If you agree that this is a sensible feature I could contribute the code.

Best wishes, 
Jürgen

**Comment by philross** [comment]
Hi @jusschwitalla ,

That is a good idea! Depending on your availability, we could also implement this on our side this week.

**Comment by jusschwitalla** [comment]
Hi Philipp,

I already did most of the work (see branch device-config in my fork). The only thing left is to check what the effect on the plots will be – probably the construction of the “Combos” has to be adapted.

Best wishes,
Jürgen

Von: Philipp ***@***.***>
Gesendet: Montag, 6. Februar 2023 09:06
An: QUARK-framework/QUARK ***@***.***>
Cc: Juergen Schwitalla ***@***.***>; Mention ***@***.***>
Betreff: Re: [QUARK-framework/QUARK] Configuration of devices (Discussion #23)


Caution: External email. Do not open attachments or click links, unless this email comes from a known sender and you know the content is safe.



Hi @jusschwitalla<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjusschwitalla&data=05%7C01%7Cjuergen.schwitalla%40atos.net%7C4128299bbb3e4f25238e08db08190d4f%7C33440fc6b7c7412cbb730e70b0198d5a%7C0%7C0%7C638112675906448843%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=0RIP5urApoQhmiHUbU6wSxMTUqqNjrrvxCRSwmaSqvM%3D&reserved=0> ,

That is a good idea! Depending on your availability, we could also implement this on our side this week.

**Comment by philross** [comment]
That is excellent @jusschwitalla, then we will take that branch as a starting point, and add we will check the effects on the plotting. 
Hopefully, we can open a pull request by the end of the week.

Best
Philipp

**Comment by Marvmann** [comment]
Thanks @jusschwitalla! Unfortunately, because of some delayed action on my part, we will only be able to open a pull request next week. Let's discuss that in the open call on Thursday. Looking forward to meeting you there!

**Comment by Marvmann** [comment]
Discussion resolved with PR#50
